### PR TITLE
bugfix flytekit-pandera

### PR DESCRIPTION
**Pull request by cosmicBboy**
Signed-off-by: Niels Bantilan <niels.bantilan@gmail.com>

# TL;DR

This PR fixes an issue where the pandera schema was not validating out-going dataframes in the `PanderaTransformer.to_literal` method. This resulted in a no-op when executing a task on a flyte backend

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

The flytekit pandera plugin was not working correctly on a flyte backend because the pandera schema was not being invoked in the `PanderaTransformer.to_literal` method. This PR fixes this by validating a dataframe output before it's written out to disk.

## Tracking Issue
NA

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/core/base_task.py [hunk 1]
```diff
@@ -524,7 +524,7 @@ def dispatch_execute(
                 try:
                     literals[k] = TypeEngine.to_literal(exec_ctx, v, py_type, literal_type)
                 except Exception as e:
-                    raise AssertionError(f"failed to convert return value for var {k}") from e
+                    raise AssertionError(f"failed to convert return value for var {k} with error: {e}") from e
```
Comment (cosmicBboy) [inline flytekit/core/base_task.py:527]: @wild-endeavor I kept `from e` because I think it'll useful when locally executing tasks

**Comment by kumare3** [comment]
@cosmicBboy do you also want to add the auto-loading feature as in the flytekitplugins-data-fsspec?

**Comment by cosmicBboy** [comment]
> @cosmicBboy do you also want to add the auto-loading feature as in the flytekitplugins-data-fsspec?

sure! how does one add the auto-loading feature?

**Comment by cosmicBboy** [comment]
> @cosmicBboy do you also want to add the auto-loading feature as in the flytekitplugins-data-fsspec?

sure! @kumare3 how does one add the auto-loading feature?

**Comment by cosmicBboy** [comment]
was able to smoke test this on sandbox, producing an error as expected:
image

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
let's do autoloading as a separate pr.
